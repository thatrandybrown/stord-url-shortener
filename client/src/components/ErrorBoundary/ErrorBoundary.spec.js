import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { ErrorBoundary } from "./";

const Broken = ({ shouldThrow }) => {
  if (shouldThrow) throw new Error("Error!");
  return <p>Success</p>;
};

beforeAll(() => {
  jest.spyOn(console, "error").mockImplementation(() => {});
});

afterAll(() => {
  console.error.mockRestore();
});

test("renders the ErrorBoundary Component with Children", () => {
  const { container, queryByText } = render(
    <ErrorBoundary>
      <Broken shouldThrow={false} />
    </ErrorBoundary>
  );

  expect(queryByText("Success")).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot();
});

test("renders the Error Component when an Error is thrown", () => {
  const { container, rerender, queryByText } = render(
    <ErrorBoundary>
      <Broken shouldThrow={false} />
    </ErrorBoundary>,
    { wrapper: MemoryRouter }
  );
  rerender(
    <ErrorBoundary>
      <Broken shouldThrow={true} />
    </ErrorBoundary>,
    { wrapper: MemoryRouter }
  );
  expect(container.firstChild).toMatchSnapshot();
  expect(queryByText("Oops!")).toBeInTheDocument();
});
