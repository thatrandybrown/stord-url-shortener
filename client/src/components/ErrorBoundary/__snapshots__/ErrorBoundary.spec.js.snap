// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Error Component when an Error is thrown 1`] = `
<section
  class="error-section"
>
  <h2
    style="font-size: 10rem;"
  >
    Oops!
  </h2>
  <p>
    Something went wrong!
  </p>
  <a
    href="/"
  >
    Navigate Home
  </a>
</section>
`;

exports[`renders the ErrorBoundary Component with Children 1`] = `
<p>
  Success
</p>
`;
